collector/benthos/output/openmeter.go

```diff
@@ -88,84 +97,82 @@ func newOpenMeterOutput(conf *service.ParsedConfig) (*openmeterOutput, error) {
 			return nil, err
 		}
 	}
+	o.client = client
 
-	return &openmeterOutput{
-		client: client,
-	}, nil
+	return o, nil
 }
 
-func (out *openmeterOutput) Connect(_ context.Context) error {
+func (o *openmeterOutput) Connect(_ context.Context) error {
 	return nil
 }
 
 // TODO: add schema validation
-func (out *openmeterOutput) WriteBatch(ctx context.Context, batch service.MessageBatch) error {
+func (o *openmeterOutput) WriteBatch(ctx context.Context, batch service.MessageBatch) error {
 	// if there is only one message use the single message endpoint
 	// otherwise use the batch endpoint
 	// if validation is enabled, try to parse the message as cloudevents first
 	//
 
-	var contentType string
-	var body io.Reader
-
-	// No need to send a batch if there is only one message
-	if len(batch) == 1 {
-		contentType = "application/cloudevents+json"
+	var err error
+	var events []any
 
-		b, err := batch[0].AsBytes()
-		if err != nil {
-			return err
+	walkFn := func(_ int, msg *service.Message) error {
+		if msg == nil {
+			return errors.New("message is nil")
 		}
 
-		body = bytes.NewReader(b)
-	} else {
-		contentType = "application/cloudevents-batch+json"
-
-		events := make([]any, 0, len(batch))
-
-		err := batch.WalkWithBatchedErrors(func(_ int, msg *service.Message) error {
-			e, err := msg.AsStructured()
-			if err != nil {
-				return err
-			}
-
-			events = append(events, e)
-
-			return nil
-		})
+		var e any
+		e, err = msg.AsStructured()
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to convert message to structed data: %w", err)
 		}
+		events = append(events, e)
 
-		var b bytes.Buffer
+		return nil
+	}
+	if err = batch.WalkWithBatchedErrors(walkFn); err != nil {
+		return fmt.Errorf("failed to process event: %w", err)
+	}
 
-		err = json.NewEncoder(&b).Encode(events)
-		if err != nil {
-			return err
-		}
+	if len(events) <= 0 {
```

Comment:
Length of a slice cannot be less than 0.

Reply:
I am aware, but its a good practice to exclude negative numers even if `len` prticualrly cannot return negative number.

Reply:
Fixed.